File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
//     + - 다 달아보면 될듯
//     BFS 느낌으로
    
    
    function DFS(index,sum){
        if(index === numbers.length){
//             끝에 다다르면 확인한다
            if(sum===target){
                answer++
            }
            return
        }
        DFS(index+1, sum-numbers[index])
        DFS(index+1,sum+numbers[index])
    }
    
    DFS(0,0)
    return answer;
}

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global answer
    answer = 0
    l1 = []
    def DFS():
        global answer
        if len(numbers) == len(l1):
            if sum(l1) == target:
                answer+=1
                return
            else:
                return
        l1.append(numbers[len(l1)])
        DFS()
        l1.pop()
        l1.append(-numbers[len(l1)])
        DFS()
        l1.pop()
    DFS()
    return answer